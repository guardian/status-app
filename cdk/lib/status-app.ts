import type { GuStackProps } from "@guardian/cdk/lib/constructs/core";
import { GuStack } from "@guardian/cdk/lib/constructs/core";
import {Tags, type App } from "aws-cdk-lib";
import {InstanceClass, InstanceSize, InstanceType, UserData} from "aws-cdk-lib/aws-ec2";
import {AccessScope} from "@guardian/cdk/lib/constants";
import {GuEc2App} from "@guardian/cdk";

export class StatusApp extends GuStack {
  constructor(scope: App, id: string, props: GuStackProps) {
    super(scope, id, props);

    const {stage: Stage} = props;
    const app = "status-app";

    const region = "eu-west-1"

    const userData = UserData.custom(`#!/bin/bash -ev
          aws --region ${region} s3 cp s3://membership-dist/${app}/${Stage}/status-app/status-app_1.0_all.deb .
          dpkg -i status-app_1.0_all.deb
          /opt/cloudwatch-logs/configure-logs application ${app} ${Stage} status-app /var/log/status-app/status-app.log
          `);

    const ec2 = new GuEc2App(this, {
      app,
      access: {
        scope: AccessScope.PUBLIC,
      },
      instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.SMALL),
      applicationPort: 9000,
      monitoringConfiguration: {noMonitoring: true},
      scaling: {minimumInstances: 1, maximumInstances: 2},
      userData,
    })
    ec2.targetGroup.healthCheck = {
      ...ec2.targetGroup.healthCheck,
      path: "/health-check"
    }

    Tags.of(ec2.autoScalingGroup).add("SystemdUnit", `${app}.service`);
  }
}
