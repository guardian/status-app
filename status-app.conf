# Upstart service configuration
env USER=status
env USER_HOME=/status
env JAR=/ophan/app.jar

env JVM_OPTIONS="-Xmx400m -XX:+PrintGCDetails -XX:+PrintGCDateStamps -Xloggc:gc.log"
env APP_OPTIONS=""

env LOGFILE=/status/stdout.log

start on runlevel [2345]
stop on runlevel [016]

# NB: setuid is only supported on recent versions of upstart - i.e.
#  on ubuntu not on amazon linux

setuid status

chdir /status

# automatically restart if the process dies
# (will enable once I've proved the rest works!)
#respawn

script
  /status/status-app-1.0/bin/status-app -mem 512 > $LOGFILE 2>&1
end script

