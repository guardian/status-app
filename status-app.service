[Unit]
Description=Status app

[Service]
Environment='MANAGEMENT_PORT=9000'
ExecStart=/status/status-app-1.0/bin/status-app -mem 512
User=status
WorkingDirectory=/status

[Install]
WantedBy=multi-user.target
