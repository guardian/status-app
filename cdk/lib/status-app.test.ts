import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import { StatusApp } from './status-app';

describe('The StatusApp stack', () => {
	it('matches the snapshot', () => {
		const app = new App();
		const stack = new StatusApp(app, 'StatusApp', {
			stack: 'ophan',
			stage: 'TEST',
		});
		const template = Template.fromStack(stack);
		expect(template.toJSON()).toMatchSnapshot();
	});
});
