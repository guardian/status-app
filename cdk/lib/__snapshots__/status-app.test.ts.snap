// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The StatusApp stack matches the snapshot 1`] = `
{
  "Metadata": {
    "gu:cdk:constructs": [
      "GuStringParameter",
      "GuStringParameter",
      "GuAllowPolicy",
      "GuAllowPolicy",
      "GuVpcParameter",
      "GuSubnetListParameter",
      "GuSubnetListParameter",
      "GuEc2App",
      "GuCertificate",
      "GuInstanceRole",
      "GuSsmSshPolicy",
      "GuDescribeEC2Policy",
      "GuLoggingStreamNameParameter",
      "GuLogShippingPolicy",
      "GuDistributionBucketParameter",
      "GuGetDistributablePolicy",
      "GuParameterStoreReadPolicy",
      "GuAmiParameter",
      "GuHttpsEgressSecurityGroup",
      "GuWazuhAccess",
      "GuAutoScalingGroup",
      "GuApplicationLoadBalancer",
      "GuApplicationTargetGroup",
      "GuHttpsApplicationListener",
      "GuDynamoTable",
    ],
    "gu:cdk:version": "TEST",
  },
  "Outputs": {
    "LoadBalancerStatusappDnsName": {
      "Description": "DNS entry for LoadBalancerStatusapp",
      "Value": {
        "Fn::GetAtt": [
          "LoadBalancerStatusapp1F67D563",
          "DNSName",
        ],
      },
    },
  },
  "Parameters": {
    "AMIStatusapp": {
      "Description": "Amazon Machine Image ID for the app status-app. Use this in conjunction with AMIgo to keep AMIs up to date.",
      "Type": "AWS::EC2::Image::Id",
    },
    "DistributionBucketName": {
      "Default": "/account/services/artifact.bucket",
      "Description": "SSM parameter containing the S3 bucket name holding distribution artifacts",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "LoggingStreamName": {
      "Default": "/account/services/logging.stream.name",
      "Description": "SSM parameter containing the Name (not ARN) on the kinesis stream",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
    "VpcId": {
      "Default": "/account/vpc/primary/id",
      "Description": "Virtual Private Cloud to run EC2 instances within. Should NOT be the account default VPC.",
      "Type": "AWS::SSM::Parameter::Value<AWS::EC2::VPC::Id>",
    },
    "hostedzoneid": {
      "Description": "ID for the hosted zone",
      "Type": "String",
    },
    "hostedzonename": {
      "Description": "DNS hosted zone for which A CNAME will be created. e.g. example.com (note, no trailing full-stop) for status.example.com. Leave empty if you don't want to add a CNAME to the status app",
      "Type": "String",
    },
    "statusappPrivateSubnets": {
      "Default": "/account/vpc/primary/subnets/private",
      "Description": "A list of private subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
    "statusappPublicSubnets": {
      "Default": "/account/vpc/primary/subnets/public",
      "Description": "A list of public subnets",
      "Type": "AWS::SSM::Parameter::Value<List<AWS::EC2::Subnet::Id>>",
    },
  },
  "Resources": {
    "AutoScalingGroupStatusappASGEE3FE15B": {
      "Properties": {
        "HealthCheckGracePeriod": 120,
        "HealthCheckType": "ELB",
        "LaunchTemplate": {
          "LaunchTemplateId": {
            "Ref": "ophanTESTstatusappB56EA109",
          },
          "Version": {
            "Fn::GetAtt": [
              "ophanTESTstatusappB56EA109",
              "LatestVersionNumber",
            ],
          },
        },
        "MaxSize": "2",
        "MetricsCollection": [
          {
            "Granularity": "1Minute",
          },
        ],
        "MinSize": "1",
        "Tags": [
          {
            "Key": "App",
            "PropagateAtLaunch": true,
            "Value": "status-app",
          },
          {
            "Key": "gu:cdk:version",
            "PropagateAtLaunch": true,
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "PropagateAtLaunch": true,
            "Value": "guardian/status-app",
          },
          {
            "Key": "LogKinesisStreamName",
            "PropagateAtLaunch": true,
            "Value": {
              "Ref": "LoggingStreamName",
            },
          },
          {
            "Key": "Stack",
            "PropagateAtLaunch": true,
            "Value": "ophan",
          },
          {
            "Key": "Stage",
            "PropagateAtLaunch": true,
            "Value": "TEST",
          },
          {
            "Key": "SystemdUnit",
            "PropagateAtLaunch": true,
            "Value": "status-app.service",
          },
        ],
        "TargetGroupARNs": [
          {
            "Ref": "TargetGroupStatusapp66994662",
          },
        ],
        "VPCZoneIdentifier": {
          "Ref": "statusappPrivateSubnets",
        },
      },
      "Type": "AWS::AutoScaling::AutoScalingGroup",
    },
    "CertificateStatusapp6FD8E329": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "DomainName": "status.ophan.co.uk",
        "DomainValidationOptions": [
          {
            "DomainName": "status.ophan.co.uk",
            "HostedZoneId": {
              "Ref": "hostedzoneid",
            },
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "status-app",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/status-app",
          },
          {
            "Key": "Name",
            "Value": "StatusApp/CertificateStatusapp",
          },
          {
            "Key": "Stack",
            "Value": "ophan",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "ValidationMethod": "DNS",
      },
      "Type": "AWS::CertificateManager::Certificate",
      "UpdateReplacePolicy": "Retain",
    },
    "ConfigTable5CD72349": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "AttributeDefinitions": [
          {
            "AttributeName": "key",
            "AttributeType": "S",
          },
        ],
        "DeletionProtectionEnabled": true,
        "KeySchema": [
          {
            "AttributeName": "key",
            "KeyType": "HASH",
          },
        ],
        "ProvisionedThroughput": {
          "ReadCapacityUnits": 1,
          "WriteCapacityUnits": 1,
        },
        "TableName": "StatusAppConfig-TEST",
        "Tags": [
          {
            "Key": "devx-backup-enabled",
            "Value": "true",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/status-app",
          },
          {
            "Key": "Stack",
            "Value": "ophan",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::DynamoDB::Table",
      "UpdateReplacePolicy": "Retain",
    },
    "DescribeEC2PolicyFF5F9295": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "autoscaling:DescribeAutoScalingInstances",
                "autoscaling:DescribeAutoScalingGroups",
                "ec2:DescribeTags",
                "ec2:DescribeInstances",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "describe-ec2-policy",
        "Roles": [
          {
            "Ref": "InstanceRoleStatusapp3C847643",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GetDistributablePolicyStatusappC309984D": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:GetObject",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:s3:::",
                    {
                      "Ref": "DistributionBucketName",
                    },
                    "/ophan/TEST/status-app/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GetDistributablePolicyStatusappC309984D",
        "Roles": [
          {
            "Ref": "InstanceRoleStatusapp3C847643",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "GuHttpsEgressSecurityGroupStatusappBB2B54EA": {
      "Properties": {
        "GroupDescription": "Allow all outbound HTTPS traffic",
        "SecurityGroupEgress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow all outbound HTTPS traffic",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "status-app",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/status-app",
          },
          {
            "Key": "Stack",
            "Value": "ophan",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "GuHttpsEgressSecurityGroupStatusappfromStatusAppLoadBalancerStatusappSecurityGroup0536BC4B9000A44B80D6": {
      "Properties": {
        "Description": "Load balancer to target",
        "FromPort": 9000,
        "GroupId": {
          "Fn::GetAtt": [
            "GuHttpsEgressSecurityGroupStatusappBB2B54EA",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": {
          "Fn::GetAtt": [
            "LoadBalancerStatusappSecurityGroupF1642DB8",
            "GroupId",
          ],
        },
        "ToPort": 9000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "GuLogShippingPolicy981BFE5A": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "kinesis:Describe*",
                "kinesis:Put*",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:kinesis:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":stream/",
                    {
                      "Ref": "LoggingStreamName",
                    },
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "GuLogShippingPolicy981BFE5A",
        "Roles": [
          {
            "Ref": "InstanceRoleStatusapp3C847643",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "InstanceRoleStatusapp3C847643": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "ec2.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "Path": "/",
        "Tags": [
          {
            "Key": "App",
            "Value": "status-app",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/status-app",
          },
          {
            "Key": "Stack",
            "Value": "ophan",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "ListenerStatusappC9B42FDB": {
      "Properties": {
        "Certificates": [
          {
            "CertificateArn": {
              "Ref": "CertificateStatusapp6FD8E329",
            },
          },
        ],
        "DefaultActions": [
          {
            "TargetGroupArn": {
              "Ref": "TargetGroupStatusapp66994662",
            },
            "Type": "forward",
          },
        ],
        "LoadBalancerArn": {
          "Ref": "LoadBalancerStatusapp1F67D563",
        },
        "Port": 443,
        "Protocol": "HTTPS",
        "SslPolicy": "ELBSecurityPolicy-TLS13-1-2-2021-06",
      },
      "Type": "AWS::ElasticLoadBalancingV2::Listener",
    },
    "LoadBalancerStatusapp1F67D563": {
      "Properties": {
        "LoadBalancerAttributes": [
          {
            "Key": "deletion_protection.enabled",
            "Value": "true",
          },
          {
            "Key": "routing.http.x_amzn_tls_version_and_cipher_suite.enabled",
            "Value": "true",
          },
          {
            "Key": "routing.http.drop_invalid_header_fields.enabled",
            "Value": "true",
          },
        ],
        "Scheme": "internet-facing",
        "SecurityGroups": [
          {
            "Fn::GetAtt": [
              "LoadBalancerStatusappSecurityGroupF1642DB8",
              "GroupId",
            ],
          },
        ],
        "Subnets": {
          "Ref": "statusappPublicSubnets",
        },
        "Tags": [
          {
            "Key": "App",
            "Value": "status-app",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/status-app",
          },
          {
            "Key": "Stack",
            "Value": "ophan",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "Type": "application",
      },
      "Type": "AWS::ElasticLoadBalancingV2::LoadBalancer",
    },
    "LoadBalancerStatusappSecurityGroupF1642DB8": {
      "Properties": {
        "GroupDescription": "Automatically created Security Group for ELB StatusAppLoadBalancerStatusappD77C3F64",
        "SecurityGroupIngress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Allow from anyone on port 443",
            "FromPort": 443,
            "IpProtocol": "tcp",
            "ToPort": 443,
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "status-app",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/status-app",
          },
          {
            "Key": "Stack",
            "Value": "ophan",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "LoadBalancerStatusappSecurityGrouptoStatusAppGuHttpsEgressSecurityGroupStatusappA0F9CBB690006ED10833": {
      "Properties": {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": {
          "Fn::GetAtt": [
            "GuHttpsEgressSecurityGroupStatusappBB2B54EA",
            "GroupId",
          ],
        },
        "FromPort": 9000,
        "GroupId": {
          "Fn::GetAtt": [
            "LoadBalancerStatusappSecurityGroupF1642DB8",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 9000,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "LoadBalancerStatusappSecurityGrouptoStatusAppWazuhSecurityGroup83D60B6C900002C2372C": {
      "Properties": {
        "Description": "Load balancer to target",
        "DestinationSecurityGroupId": {
          "Fn::GetAtt": [
            "WazuhSecurityGroup",
            "GroupId",
          ],
        },
        "FromPort": 9000,
        "GroupId": {
          "Fn::GetAtt": [
            "LoadBalancerStatusappSecurityGroupF1642DB8",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "ToPort": 9000,
      },
      "Type": "AWS::EC2::SecurityGroupEgress",
    },
    "ParameterStoreReadStatusapp44EB488E": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "ssm:GetParametersByPath",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/ophan/status-app",
                  ],
                ],
              },
            },
            {
              "Action": [
                "ssm:GetParameters",
                "ssm:GetParameter",
              ],
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:ssm:",
                    {
                      "Ref": "AWS::Region",
                    },
                    ":",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":parameter/TEST/ophan/status-app/*",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "parameter-store-read-policy",
        "Roles": [
          {
            "Ref": "InstanceRoleStatusapp3C847643",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "SsmSshPolicy4CFC977E": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "ec2messages:AcknowledgeMessage",
                "ec2messages:DeleteMessage",
                "ec2messages:FailMessage",
                "ec2messages:GetEndpoint",
                "ec2messages:GetMessages",
                "ec2messages:SendReply",
                "ssm:UpdateInstanceInformation",
                "ssm:ListInstanceAssociations",
                "ssm:DescribeInstanceProperties",
                "ssm:DescribeDocumentParameters",
                "ssmmessages:CreateControlChannel",
                "ssmmessages:CreateDataChannel",
                "ssmmessages:OpenControlChannel",
                "ssmmessages:OpenDataChannel",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "ssm-ssh-policy",
        "Roles": [
          {
            "Ref": "InstanceRoleStatusapp3C847643",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "TargetGroupStatusapp66994662": {
      "Properties": {
        "HealthCheckIntervalSeconds": 10,
        "HealthCheckPath": "/management/healthcheck",
        "HealthCheckProtocol": "HTTP",
        "HealthCheckTimeoutSeconds": 5,
        "HealthyThresholdCount": 2,
        "Port": 9000,
        "Protocol": "HTTP",
        "Tags": [
          {
            "Key": "App",
            "Value": "status-app",
          },
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/status-app",
          },
          {
            "Key": "Stack",
            "Value": "ophan",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "TargetGroupAttributes": [
          {
            "Key": "deregistration_delay.timeout_seconds",
            "Value": "30",
          },
          {
            "Key": "stickiness.enabled",
            "Value": "false",
          },
        ],
        "TargetType": "instance",
        "UnhealthyThresholdCount": 2,
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
    },
    "WazuhSecurityGroup": {
      "Properties": {
        "GroupDescription": "Allow outbound traffic from wazuh agent to manager",
        "SecurityGroupEgress": [
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Wazuh event logging",
            "FromPort": 1514,
            "IpProtocol": "tcp",
            "ToPort": 1514,
          },
          {
            "CidrIp": "0.0.0.0/0",
            "Description": "Wazuh agent registration",
            "FromPort": 1515,
            "IpProtocol": "tcp",
            "ToPort": 1515,
          },
        ],
        "Tags": [
          {
            "Key": "gu:cdk:version",
            "Value": "TEST",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/status-app",
          },
          {
            "Key": "Stack",
            "Value": "ophan",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
        "VpcId": {
          "Ref": "VpcId",
        },
      },
      "Type": "AWS::EC2::SecurityGroup",
    },
    "WazuhSecurityGroupfromStatusAppLoadBalancerStatusappSecurityGroup0536BC4B9000525BF445": {
      "Properties": {
        "Description": "Load balancer to target",
        "FromPort": 9000,
        "GroupId": {
          "Fn::GetAtt": [
            "WazuhSecurityGroup",
            "GroupId",
          ],
        },
        "IpProtocol": "tcp",
        "SourceSecurityGroupId": {
          "Fn::GetAtt": [
            "LoadBalancerStatusappSecurityGroupF1642DB8",
            "GroupId",
          ],
        },
        "ToPort": 9000,
      },
      "Type": "AWS::EC2::SecurityGroupIngress",
    },
    "cnamerecord": {
      "Properties": {
        "Comment": "CNAME for status app",
        "HostedZoneName": {
          "Fn::Join": [
            "",
            [
              {
                "Ref": "hostedzonename",
              },
              ".",
            ],
          ],
        },
        "Name": {
          "Fn::Join": [
            "",
            [
              "status.",
              {
                "Ref": "hostedzonename",
              },
            ],
          ],
        },
        "ResourceRecords": [
          {
            "Fn::GetAtt": [
              "LoadBalancerStatusapp1F67D563",
              "DNSName",
            ],
          },
        ],
        "TTL": "900",
        "Type": "CNAME",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "dynamoaccess94F8BF99": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "dynamodb:GetItem",
              "Effect": "Allow",
              "Resource": {
                "Fn::Join": [
                  "",
                  [
                    "arn:aws:dynamodb:eu-west-1:",
                    {
                      "Ref": "AWS::AccountId",
                    },
                    ":table/StatusAppConfig",
                  ],
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "dynamoaccess94F8BF99",
        "Roles": [
          {
            "Ref": "InstanceRoleStatusapp3C847643",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "ophanTESTstatusappB56EA109": {
      "DependsOn": [
        "InstanceRoleStatusapp3C847643",
      ],
      "Properties": {
        "LaunchTemplateData": {
          "IamInstanceProfile": {
            "Arn": {
              "Fn::GetAtt": [
                "ophanTESTstatusappProfileF1C8DBF4",
                "Arn",
              ],
            },
          },
          "ImageId": {
            "Ref": "AMIStatusapp",
          },
          "InstanceType": "t4g.small",
          "MetadataOptions": {
            "HttpTokens": "required",
            "InstanceMetadataTags": "enabled",
          },
          "SecurityGroupIds": [
            {
              "Fn::GetAtt": [
                "GuHttpsEgressSecurityGroupStatusappBB2B54EA",
                "GroupId",
              ],
            },
            {
              "Fn::GetAtt": [
                "WazuhSecurityGroup",
                "GroupId",
              ],
            },
          ],
          "TagSpecifications": [
            {
              "ResourceType": "instance",
              "Tags": [
                {
                  "Key": "App",
                  "Value": "status-app",
                },
                {
                  "Key": "gu:cdk:version",
                  "Value": "TEST",
                },
                {
                  "Key": "gu:repo",
                  "Value": "guardian/status-app",
                },
                {
                  "Key": "Name",
                  "Value": "StatusApp/ophan-TEST-status-app",
                },
                {
                  "Key": "Stack",
                  "Value": "ophan",
                },
                {
                  "Key": "Stage",
                  "Value": "TEST",
                },
              ],
            },
            {
              "ResourceType": "volume",
              "Tags": [
                {
                  "Key": "App",
                  "Value": "status-app",
                },
                {
                  "Key": "gu:cdk:version",
                  "Value": "TEST",
                },
                {
                  "Key": "gu:repo",
                  "Value": "guardian/status-app",
                },
                {
                  "Key": "Name",
                  "Value": "StatusApp/ophan-TEST-status-app",
                },
                {
                  "Key": "Stack",
                  "Value": "ophan",
                },
                {
                  "Key": "Stage",
                  "Value": "TEST",
                },
              ],
            },
          ],
          "UserData": {
            "Fn::Base64": "#!/bin/bash -ev
          aws --region eu-west-1 s3 cp s3://ophan-dist/ophan/TEST/status-app/status-app_1.0_all.deb .
          dpkg -i status-app_1.0_all.deb
          /opt/cloudwatch-logs/configure-logs application ophan TEST status-app /var/log/status-app/status-app.log
         ",
          },
        },
        "TagSpecifications": [
          {
            "ResourceType": "launch-template",
            "Tags": [
              {
                "Key": "App",
                "Value": "status-app",
              },
              {
                "Key": "gu:cdk:version",
                "Value": "TEST",
              },
              {
                "Key": "gu:repo",
                "Value": "guardian/status-app",
              },
              {
                "Key": "Name",
                "Value": "StatusApp/ophan-TEST-status-app",
              },
              {
                "Key": "Stack",
                "Value": "ophan",
              },
              {
                "Key": "Stage",
                "Value": "TEST",
              },
            ],
          },
        ],
      },
      "Type": "AWS::EC2::LaunchTemplate",
    },
    "ophanTESTstatusappProfileF1C8DBF4": {
      "Properties": {
        "Roles": [
          {
            "Ref": "InstanceRoleStatusapp3C847643",
          },
        ],
      },
      "Type": "AWS::IAM::InstanceProfile",
    },
    "readmetadata4512F3FD": {
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": [
                "ec2:describe*",
                "autoscaling:Describe*",
                "elasticloadbalancing:Describe*",
                "cloudwatch:Get*",
                "sqs:ListQueues",
              ],
              "Effect": "Allow",
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "readmetadata4512F3FD",
        "Roles": [
          {
            "Ref": "InstanceRoleStatusapp3C847643",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
  },
}
`;
